HealthMonitoring.SelfHost/Content/Scripts/homeController.js

```diff
@@ -122,26 +196,135 @@
+                Unhealthy: item.Status === "unhealthy" ? groupsInfo[item.Group].Unhealthy + 1 : groupsInfo[item.Group].Unhealthy,
+                TimedOut: item.Status === "timedOut" ? groupsInfo[item.Group].TimedOut + 1 : groupsInfo[item.Group].TimedOut,
+                Name: item.Group,
+                TagInfo: groupsInfo[item.Group].TagInfo,
+                LongestResponseTime: getLongestResponseTime(),
+                ShortestResponseTime: getShortestResponseTime(),
+                LastCheckUtc: item.LastCheckUtc
+            }
+        } else {
+            groupsInfo[item.Group] = {
+                Count: 1,
+                NotRun: item.Status === "notRun" ? 1 : 0,
+                NotExists: item.Status === "notExists" ? 1 : 0,
+                Offline: item.Status === "offline" ? 1 : 0,
+                Healthy: item.Status === "healthy" ? 1 : 0,
+                Faulty: item.Status === "faulty" ? 1 : 0,
+                Unhealthy: item.Status === "unhealthy" ? 1 : 0,
+                TimedOut: item.Status === "timedOut" ? 1 : 0,
+                Name: item.Group,
+                TagInfo: [],
+                LongestResponseTime: item.LastResponseTime,
+                ShortestResponseTime: item.LastResponseTime,
+                LastCheckUtc: item.LastCheckUtc
+            }
+        }
+
+        angular.forEach(getNewTags(), addTagToGroupInfoIfDoesntExist);
+        
+        function getNewTags() {
+            var newTags = [];
+
+            angular.forEach(item.Tags, collateNewTags);
+
+            return newTags;
+
+            function collateNewTags(newTag) {
+                newTags.push(newTag);
+            }
+        }
+
+        function addTagToGroupInfoIfDoesntExist(newTag) {
+            if (isUndefinedOrNullOrEmptyOrHasWhiteSpaces(newTag)) {
+                return;
+            }
+
+            var currentTags = [];
+            
+            groupsInfo[item.Group].TagInfo.forEach(collateCurrentTags);
+            
+            if (currentTags.indexOf(newTag) === -1) {
+                groupsInfo[item.Group].TagInfo.push({
+                    Tag: newTag,
+                    Id: item.Id
+                });
+            }
+
+            function collateCurrentTags(tagInfo) {
+                currentTags.push(tagInfo.Tag);
+            }
+        }
+
+        function getLongestResponseTime() {
+            return groupsInfo[item.Group].LongestResponseTime > item.LastResponseTime
+                ? groupsInfo[item.Group].LongestResponseTime
+                : item.LastResponseTime;
+        }
+
+        function getShortestResponseTime() {
+            return groupsInfo[item.Group].ShortestResponseTime < item.LastResponseTime
+                ? groupsInfo[item.Group].ShortestResponseTime
+                : item.LastResponseTime;
+        }
+        
+    }
+}
+
+function lowercaseFirstLetter(string) {
```

Comment:
![code](https://cloud.githubusercontent.com/assets/19727177/23909476/8a46f3b0-08df-11e7-811f-00fbd2b8243d.JPG)
Sorry but I still cant find any usage of this method